src/crypto/hash.cpp

```diff
@@ -22,6 +22,15 @@ namespace crypto
     return r;
   }
 
+  std::vector<uint8_t> SHA256(const uint8_t* data, size_t len)
```

Comment:
Should the `SHA256(const std::vector<uint8_t>&)` function above call into this new one to avoid code duplication?

Reply:
I nearly did that, but it involves creating an additional CBuffer that may or may not be optimised out, and it's very little code, so I decided against it. I could be convinced otherwise.

Reply:
No?
```
  std::vector<uint8_t> SHA256(const std::vector<uint8_t>& data)
  {
    return SHA256(data.data(), data.size());
  }